src/com/xilinx/rapidwright/router/RouteThruHelper.java

```diff
@@ -194,11 +195,19 @@ public static boolean isRouteThruPIPAvailable(Design design, Wire start, Wire en
         if (!isRouteThruSitePinAvailable(design, inPin)) {
             return false;
         }
+
         SiteInst siteInst = design.getSiteInstFromSite(inPin.getSite());
         if (siteInst != null) {
             for (BELPin sink : inPin.getBELPin().getSiteConns()) {
-                Cell cellCollision = siteInst.getCell(sink.getBEL());
+                BEL sinkBEL = sink.getBEL();
+                if (sinkBEL.getName().charAt(0) != inPin.getPinName().charAt(0)) {
+                    continue;
+                }
+                Cell cellCollision = siteInst.getCell(sinkBEL);
                 if (cellCollision != null) {
+                    // Ignore BELs that don't share the same LUT letter
+                    // Specifically, this is to prevent H[1-6] inputs on SLICEM sites
+                    // -- which also drive [A-G].WA[1-6] -- from considering [A-G]LUT[56]
```

Comment:
```suggestion
                    // Ignore BELs that don't share the same LUT letter
                    // Specifically, this is to prevent H[1-6] inputs on SLICEM sites
                    // -- which also drive [A-G].WA[1-6] -- from considering [A-G]LUT[56]
                    continue;
                }
                Cell cellCollision = siteInst.getCell(sinkBEL);
                if (cellCollision != null) {
```